handlers/meme.go

```diff
@@ -0,0 +1,229 @@
+	dirName := "uploads"
+	file, header, err := r.FormFile("file")
+	if err != nil {
+		http.Error(w, "Unable to parse file", http.StatusBadRequest)
+		return
+	}
+	defer file.Close()
+
+	// Check if uploads directory exists or create it
+	CreateUploadsDirectory(dirName)
+
+	// Saving the file
+	dst, err := os.Create(dirName + "/" + header.Filename)
+
+	if err != nil {
+		http.Error(w, "Unable to save file 1", http.StatusInternalServerError)
+		return
+	}
+
+	defer dst.Close()
+
+	_, err = io.Copy(dst, file)
+	if err != nil {
+		http.Error(w, "Unable to save file 2", http.StatusInternalServerError)
+		return
+	}
+
+	challenge := GetMemeChallenge()
+	signer := SignChallenge(challenge.Challenge)
+	mErr, mToken := GetMemeToken(challenge.Id, signer.Response.Sig)
+
+	if mErr != "" {
+		msg := "Could not get meme token"
+		fmt.Println(msg, mErr)
+		w.WriteHeader(http.StatusNoContent)
+		json.NewEncoder(w).Encode(msg)
+	} else {
+		memeImgUrl := UploadMemeImage(file, mToken.Token, header.Filename)
+		if memeImgUrl == "" {
+			msg := "Could not get meme image"
+			fmt.Println(msg)
+			w.WriteHeader(http.StatusNoContent)
+			json.NewEncoder(w).Encode(msg)
+		} else {
+			w.WriteHeader(http.StatusOK)
+			json.NewEncoder(w).Encode(memeImgUrl)
+		}
+	}
+}
+
+func GetMemeChallenge() db.MemeChallenge {
+	memeChallenge := db.MemeChallenge{}
+
+	url := fmt.Sprintf("%s/ask", config.MemeUrl)
+
+	client := &http.Client{}
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+
+	req.Header.Set("Content-Type", "application/json")
+	res, _ := client.Do(req)
+
+	if err != nil {
+		log.Printf("Request Failed: %s", err)
+	}
+
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+
+	// Unmarshal result
+	err = json.Unmarshal(body, &memeChallenge)
+
+	if err != nil {
+		log.Printf("Reading Invoice body failed: %s", err)
+	}
+
+	return memeChallenge
+}
+
+func SignChallenge(challenge string) db.RelaySignerResponse {
+	url := fmt.Sprintf("%s/signer/%s", config.RelayUrl, challenge)
+
+	client := &http.Client{}
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+
+	req.Header.Set("x-user-token", config.RelayAuthKey)
+	req.Header.Set("Content-Type", "application/json")
+	res, _ := client.Do(req)
+
+	if err != nil {
+		log.Printf("Request Failed: %s", err)
+	}
+
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+
+	signerResponse := db.RelaySignerResponse{}
+
+	// Unmarshal result
+	err = json.Unmarshal(body, &signerResponse)
+
+	if err != nil {
+		log.Printf("Reading Challenge body failed: %s", err)
+	}
+
+	return signerResponse
+}
+
+func GetMemeToken(id string, sig string) (string, db.MemeTokenSuccess) {
+	memeUrl := fmt.Sprintf("%s/verify", config.MemeUrl)
+
+	formData := url.Values{
+		"id":     {id},
+		"sig":    {sig},
+		"pubkey": {config.RelayNodeKey},
+	}
+
+	res, err := http.PostForm(memeUrl, formData)
+
+	if err != nil {
+		log.Printf("Request Failed: %s", err)
+		return "", db.MemeTokenSuccess{}
```

Comment:
are we able to return nil here 
```suggestion
		return nil, db.MemeTokenSuccess{}
```